internetarchive/openlibrary
# Create aggregate observations component for book pages

Closes #4867


Creates a new component for book pages which displays patron observation statistics for the work.  A new "Reader Observations" tab has been added above the editions table to provide quick access to the new component.

### Technical

The book notes modal, `UserMetadata.html`, has been modified to accept a unique ID as a parameter.  This allows multiple modals to be displayed on a single page.  This may not be absolutely necessary for this PR, but it will be needed for the upcoming consolidated book notes view PR.

A modal link is included at the bottom of the new component, prompting patrons to leave their own observations about the work.  This link opens the login page for folks who are not already logged in.

Observation counts will not include deleted observation types and values.  The assumption about deleted types and values is that a boolean 'deleted' entry will be included at the observation or value level if an entire observation or a single value is deleted.

Only observations that have been submitted are displayed in the new component.  If nobody has submitted a "pacing" observation for a work, the "pacing" observation will be absent from the component.  If a specific value for an observation has not been submitted for a work, that value will also not be present in the component.

Observations in the component are listed by type ID, ascending.  Values for each observation are listed by most submitted to least submitted.

Statistics in the observations component are not automatically refreshed when new observations are submitted.  The page must be refreshed before the changes are visible.

Feature flags are used for the new component and tab.  One must be a member of the "beta-testers" group to see them.

### Testing

While logged in as a beta-tester:

1. Navigate to a book page.
2. Ensure that the "Reader Observations" tab exists, and that clicking it links to the new component.
3. Ensure that the statistics in the component are correct (it may be helpful to query the `observations` table to verify).
4. Ensure that the modal link in the observations component opens the book notes modal.

### Screenshot

Observations component (observations for work exist):

![component-has_observations](https://user-images.githubusercontent.com/28732543/113199829-8eb3c480-9235-11eb-8525-b4cf1c94287a.png)

Observations component (no observations submitted for work):

![component-no_observations](https://user-images.githubusercontent.com/28732543/113199874-9f643a80-9235-11eb-9585-1294b5655de4.png)


### Stakeholders

@mekarpeles 
@cdrini
